### PR TITLE
Add cell value scaling functions

### LINKED ISSUE #218: Add rf_clip and rf_rescale and rf_normalize
@vpipkt Interested in your thoughts on this.

### PR DESCRIPTION
**Pull request by vpipkt**
Fix #218 

Functions we would like to include here:
* local min
* local max
* local clip 
* rescale (now much more compelling with #429 merged)
* standardize 
* ifcell?

Initial commit is just some of the functions. Still a few more to go.

### CODE REVIEW

**Comment by vpipkt** [comment]
#443

File: core/src/main/scala/org/locationtech/rasterframes/expressions/localops/Clip.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+@ExpressionDescription(
+  usage = "_FUNC_(tile, min, max) - Return the tile with its values clipped to a range defined by min and max," +
+            " doing so cellwise if min or max are tile type",
+  arguments = """
+      Arguments:
+        * tile - the tile to operate on
+        * min - scalar or tile setting the minimum value for each cell
+        * max - scalar or tile setting the maximum value for each cell"""
+)
+case class Clip(left: Expression, middle: Expression, right: Expression)
```
Comment (metasim) [inline core/src/main/scala/org/locationtech/rasterframes/expressions/localops/Clip.scala]: There's already precident for calling this kind of operation "clamp". 
https://github.com/locationtech/rasterframes/search?q=clamp&unscoped_q=clamp

A clip in imagery usually implies trimming the pixel edges, where as a clamp is on the pixel values.
https://imagemagick.org/script/command-line-options.php#clamp

So my preference here (sorry) is to keep with that nomenclature.

File: core/src/main/scala/org/locationtech/rasterframes/expressions/localops/Max.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    * tile - left-hand-side tile
+    * rhs  - a tile or scalar value""",
+  examples = """
+  Examples:
+    > SELECT _FUNC_(tile, 1.5);
+       ...
+    > SELECT _FUNC_(tile1, tile2);
+       ..."""
+)
+case class Max(left: Expression, right:Expression) extends BinaryLocalRasterOp with CodegenFallback {
```
Comment (metasim) [inline core/src/main/scala/org/locationtech/rasterframes/expressions/localops/Max.scala]: wow... didn't realize we didn't have this.

File: core/src/test/scala/org/locationtech/rasterframes/functions/TileFunctionsSpec.scala [hunk 3]
```diff
@@ -210,6 +210,145 @@ class TileFunctionsSpec extends TestEnvironment with RasterMatchers {
...
+        )
+      )
+      .first()
+
+      result should be (true)
+
+    }
+  }
+
+  describe("standardize and rescale") {
```
Comment (metasim) [inline core/src/test/scala/org/locationtech/rasterframes/functions/TileFunctionsSpec.scala]: I wonder if these should go in the `StatFunctionsSpec` file....

**Pull request merged**